scripts/build-on-ci.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+set -u
+
+time rm -rf ./node_modules
+time npm install
+npm run build
+time rm -rf ./node_modules
+time npm install --production
```

Comment:
This double install process is what need to be happening on the CI. Since the CI should be doing the building of the project, it will need to run `npm install` without the production flag set. This way it can run `npm run build`. Once that runs, it needs to vendor the dependencies for the deployment, but `devDependencies` shouldn't exist in the `node_modules` directory. Because of this, you'll need to remove them and then reinstall them with the `--production` flag.

I would make sure that your CI isn't running with the `NODE_ENV=production` nor `NPM_CONFIG_PRODUCTION=true` variables set like that.